### PR TITLE
Add PR build validation

### PR DESCRIPTION
**Pull request by gregg-miskelly**
This PR adds back PR validation to the ConcordExtensibilitySamples repo.

Changes:
1. Added new PR .yml that builds the projects in this solution. It does so by executing the build script.
2. Eliminate BuildAndTest.proj in favor of building through build.cmd. BuildAndTest.proj was making it tricky to track down failures because it wasn't acting exactly the same as running each solution independently.
3. Removed the old build files

Testing:
- Successfully completed PR builds

### CODE REVIEW

**Comment by plnelson** [comment]
Do we need to update readme.md as well to fix the build status icons?

**Comment by gregg-miskelly** [comment]
> Do we need to update readme.md as well to fix the build status icons?

Yes. Sorry, this PR should really be a draft. I haven't gotten it to work yet.

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  windows_build:
+    strategy:
+      matrix:
+        configuration: [Debug, Release]
+
+    runs-on: windows-2019
```
Comment (WardenGnaw) [inline .github/workflows/pull-request.yml:13]: Did you want to run specifically on window-2019 and we have to manually upgrade to 2022? If we use `windows-latest`, they will move us over automatically.

Latest currently runs on `windows-2019`.

```suggestion
    runs-on: windows-latest
```

**Review by WardenGnaw — APPROVED** [review]

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (gregg-miskelly) [reply .github/workflows/pull-request.yml:13]: Yes, for this PR, we need to run on VS 2019. In the next PR, we will move to VS2022.

**Pull request merged**